include/PruManager.h

```diff
@@ -0,0 +1,78 @@
+/* 
+ * PruManager.h
+ *
+ * includes the classes required for introducing RProc functionality
+ * but at the same time still support Mmap and libprussdrv
+ *
+ *	Created on: Jul 3, 2021
+ *		Author: Dhruva Gole
+ */
+
+#include <string>
+
+#if ENABLE_PRU_UIO == 1
+#include <prussdrv.h>
+#endif
+
+#include <map>
+#include "Mmap.h"
+
+class PruManager
+{
+// expose parameters for the relevant paths
+	public:
+	unsigned int pru_num, verbose;
+	std::string state;
+	PruManager();
+	virtual void readstate() = 0;
```

Comment:
I don't think it's a good design decision to have `state` as a public member  variable and a `void readstate()` method. I'd much prefer:
```
typedef enum { kRunning, KState} State;
virtual State getState();
```

Reply:
> I don't think it's a good design decision to have state as a public member variable
Oh well the reason for that is that UIO and RPRoc have different ways of reading state. in UIO there isn't exactly a direct method, hence we had discussed in meet to maintain a certain local variable that gets toggled when UIO start is called.
And then for rproc we just read the contents of "state"... 
So since these both have different approaches, but to keep the same function name we had made it virtual in the base class. The global variable is because if it were private then the child would not inherit it and so would not be able to read it if needed anywhere.

Reply:
it should be `protected` (instead of `public` or `private`) for the child to be able to access it. Make it a `typedef enum`, no point in having it a string.

Reply:
> Make it a typedef

okay if protected, then it can be done. Then I will just rename `readstate` to `getState` and keep the function implementations the same, just change the string part to the new `typedef`